### PR TITLE
DSD-3581: update required provider for deprecated object

### PR DESCRIPTION
**Pull request by creynolds15**
BREAKING CHANGE: No longer uses deprecated object

### CODE REVIEW

File: versions.tf
Comment (souza-dan) [inline versions.tf:6]: There were some s3 changes that they reverted in the low 4 versions which is why it's like this. We're probably compatible with these changes regardless, but it may be worth being a bit more strict in saying that we should only allow above 4.9.0 to get past those versions.
```suggestion
      version = ">= 4.9.0"
```

We may also want to put an upper limit for when there's inevitably a breaking change in the provider:
```suggestion
      version = ">=4.9.0, <5.0.0"
```

**Review by souza-dan — COMMENTED** [review]
Overall looks great!

**Comment by tparikh** [comment]
@creynolds15 What's the status of this PR?

**Comment by creynolds15** [comment]
> @creynolds15 What's the status of this PR?

Good to merge once it gets approval.

**Comment by tparikh** [comment]
@souza-dan Mind reviewing the PR and approving it if it looks good. Thanks!

File: modules/aws-emr-ports/versions.tf
Comment (souza-dan) [inline modules/aws-emr-ports/versions.tf:8]: The ports module doesn't actually require aws.

It's essentially just a keeper of lists of port numbers. No real resources required.

File: modules/aws-emr-config/versions.tf
Comment (souza-dan) [inline modules/aws-emr-config/versions.tf:6]: I think we'd want to be consistent to say `>=4.9.0` to match the version at the root. If you use the root, the constraint would put you at a version >=4.9.0, but using this nested module directly wouldn't enforce that.

File: examples/ephemeral-spark/versions.tf
Comment (souza-dan) [inline examples/ephemeral-spark/versions.tf:1]: Do we need these version files in the examples? This should just get inferred by the versions of the modules we're using.

Comment (creynolds15) [reply examples/ephemeral-spark/versions.tf:1]: The tflint fails if you do not include the version files in the examples.

Comment (souza-dan) [inline examples/ephemeral-spark/versions.tf:1]: Probably fine, but it may interfere with terratest test examples, since I believe they use these examples.

File: modules/aws-emr-config/versions.tf
Comment (souza-dan) [inline modules/aws-emr-config/versions.tf:4]: I'm not sure this was duplicate, since you can use the nested modules directly and wouldn't have the provider requirements from the root module.

In fact, the root module doesn't technically need the aws provider, although it is nice to specify because it'll generate the requirements tables in the README.

Make sure to run `make terraform/docs` to get the updated README.

Comment (creynolds15) [reply modules/aws-emr-config/versions.tf:4]: @souza-dan it looks like the makefile does not authenticate with Github SSO: `ERROR: The 'Datatamer' organization has enabled or enforced SAML SSO.
To access this repository, you must use the HTTPS remote with a personal access token or SSH with an SSH key and passphrase that has been authorized for this organization.`

Is there a way I should be authenticating?

Comment (souza-dan) [reply modules/aws-emr-config/versions.tf:4]: I recommend setting up your ssh key in github. I think these are the two docs to follow to get that working:
https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account
https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-saml-single-sign-on/authorizing-an-ssh-key-for-use-with-saml-single-sign-on

Then check out the repo again:
`git clone git@github.com:Datatamer/terraform-aws-emr.git`

and the `make terraform/docs` should work. I just tried it out, so if you're stuck let me know and I can push a commit with it applied.

File: README.md
Comment (souza-dan) [inline README.md:1]: It looks like this changed from `<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->`	 to `<!-- BEGIN_TF_DOCS -->` in the latest docker images, which is what's wiping out the non generated parts of the doc. 

I tried replacing it with the new tag and running it locally, but it's still getting wiped out.

I'll ask about it internally, but in the meantime can you put back the docs outside of the tags?

Comment (creynolds15) [reply README.md:1]: What do you mean by back the docs outside of the tags?

**Review by souza-dan — APPROVED** [review]
LGTM, sorry this PR involved so much yak shaving!

**Pull request merged**
